pkg/replicas/replicas.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package replicas implements a scheme to replicate chunks
+// in such a way that
+// - the replicas are optimally dispersed to aid cross-neighbourhood redundancy
+// - the replicas addresses can be deduced by retrievers only knowing the address
+// of the original content addressed chunk
+// - no new chunk validation rules are introduced
+package replicas
+
+import (
+	"context"
+	"encoding/hex"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/file/redundancy"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type redundancyLevelType struct{}
+
+var (
+	// redundancyLevel is the context key for the redundancy level
+	redundancyLevel redundancyLevelType
+	// RetryInterval is the duration between successive additional requests
+	RetryInterval = 300 * time.Millisecond
+	//
+	// counts of replicas used for levels of increasing security
+	// the actual number of replicas needed to keep the error rate below 1/10^6
+	// for the five levels of redundancy are 0, 2, 4, 5, 19
+	// we use an approximation as the successive powers of 2
+	counts     = [5]int{0, 2, 4, 8, 16}
+	sums       = [5]int{0, 2, 6, 14, 30}
+	privKey, _ = crypto.DecodeSecp256k1PrivateKey(append([]byte{1}, make([]byte, 31)...))
+	signer     = crypto.NewDefaultSigner(privKey)
+	owner, _   = hex.DecodeString("dc5b20847f43d67928f49cd4f85d696b5a7617b5")
```

Comment:
Where is this hex string coming from?

Reply:
private key `[]byte{1}` should result in this address

Reply:
Please add this as a comment.

Reply:
resolved in #4522